### ISSUE TITLE
Docker container failing on latest build

### ISSUE THREAD

**Issue by Anjali05** [google/gvisor#721]
Hi, 
I have two runsc setup, one with the latest and one that was released in March. I am trying to run the CloudSuite workload with their provided images, I tried this: https://github.com/parsa-epfl/cloudsuite/blob/master/docs/benchmarks/in-memory-analytics.md on both my setup. Although the container runs properly on my old setup, its failing on the latest release. I checked the log files, this is what the error looks like: 
**1) .gofer logs:**
![gofer](https://user-images.githubusercontent.com/8316863/63281116-a8c3ad80-c271-11e9-88af-8f0e354d75a8.png)

**2) .boot logs**

![boot-1](https://user-images.githubusercontent.com/8316863/63281149-b5e09c80-c271-11e9-9fcc-33cecb474620.png)
![boot2](https://user-images.githubusercontent.com/8316863/63281159-b9742380-c271-11e9-8a22-e1ad12f1dabf.png)
![boot3](https://user-images.githubusercontent.com/8316863/63281167-bd07aa80-c271-11e9-8f0f-3e2939d639de.png)

It looks like some memory error to me.

**Comment by hbhasker** [comment]
Could you enable --strace in runtimeArgs in etc/docker/daemon.json and out the logs here.

**Comment by Anjali05** [comment]
@hbhasker I have enabled. Do I need to look at a specific file or trace the sentry?

**Comment by fvoznika** [comment]
I can repro the problem lcoally. The regression happened between `20190518_RC00` and `20190615_RC00`. We can bisect the changes in between and find the culprit.

**Comment by fvoznika** [comment]
The regression was added in 9cdae51feca5cee9faa198161b92a0aeece52d6c, not sure yet what's broken.

**Comment by hbhasker** [comment]
Splice is currently broken. There are a couple of issues with how it
handles blocking behaviour of the underlying fd. We should disable the
syscall for now till we fix the issue.

**Comment by fvoznika** [comment]
I initially though of splice too, but the workload doesn't call it. The problem is an inadvertently change in sendfile error code when the output file is in append mode. It should return `EINVAL`, but the refactor changed it to [`EBADF`](https://github.com/google/gvisor/blob/cf8a689be7fdb539bb789ebd52ce19432776f16d/pkg/sentry/syscalls/linux/sys_splice.go#L109). I'll send a fix shortly.

**Comment by Anjali05** [comment]
@fvoznika What was the issue exactly?

**Comment by fvoznika** [comment]
When `sendfile(2)` is called with an output file that is open in append mode, the call should fail with `EINVAL`, but in gVisor it fails with `EBADF`. The application is probably not expecting `EBADF` and fails.

**Comment by Anjali05** [comment]
I tried running another workload from the suite, this one: https://github.com/parsa-epfl/cloudsuite/blob/master/docs/benchmarks/data-caching.md. It uses a user-defined bridge network, although the container runs fine with runc, it fails with runsc giving a DNS error on the client-side. I tried running normal Ubuntu container with user-defined network and then ran `apt update` but it's not able to connect to the archive site. I am wondering if there is any other network setting I need to define in runsc if I am using user-defined bridge network, the goal is to make two containers communicate with each other on the same host.
